src/ui/views/TradeFilter/TradeFilter.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from "react";
+import { POSITIONS, SKILLS } from "../../../common/constants";
+import type FilterItem from "../../../worker/core/trade/filterItem";
+import FilterGroup from "./FilterGroup";
+
+interface tradeFilterProps {
+	setFilters: (filters: filterType) => void;
+	filters: filterType;
+}
+export type filterType = {
+	[key: string]: FilterItem;
+};
+
+const POS = POSITIONS.filter(
+	pos => !["GF", "FC", "F", "G", "KR", "PR"].includes(pos),
+);
+
+const SKILL = Object.values(SKILLS).map(skill => skill.label);
+
+const TradeFilter = (props: tradeFilterProps) => {
+	const [filters, setFilters] = useState<filterType>(props.filters);
+	const [balancedOnly, setBalancedOnly] = useState(false);
+	const [salaryCap, setSalaryCap] = useState("");
+
+	const updateFilters = (type: string, value: string[]) => {
+		if (value !== filters[type].filterData) {
+			filters[type].update(value);
+			setFilters({ ...filters });
+			props.setFilters(filters);
+		}
+	};
+
+	const toggleBalancedOnly = () => {
+		setBalancedOnly(filters.salaryCap.filterData !== "-1");
+		if (filters.salaryCap.filterData !== "-1") filters.salaryCap.update("-1");
+		else filters.salaryCap.update("");
```

Comment:
Why are you storing this as "" and "-1" rather than true and false?

Reply:
Oh, it treats those as magic values in an otherwise numeric value? Seems confusing, maybe I am missing something...

Reply:
yeah I should revisit this, there's essentially three states: Unset, set, and "balanced trade" -- didn't mean to keep it like this hah. Also some complexity with validating: the event will pass in a string for a number but I decided at some point that it would be better to cast in filtering step... but like I said probably a way that makes more sense